src/datatype/mod.rs

```diff
@@ -32,3 +33,11 @@ pub mod text_vecf16;
 pub mod text_vecf32;
 pub mod text_veci8;
 pub mod typmod;
+
+use pgrx::Internal;
+
+fn get_mut_internal<T>(internal: &mut Option<Internal>) -> Option<&mut T> {
```

Comment:
It cannot be marked safe.

Reply:
May I mark this function unsafe? Inline it would make code below not so readable?
```rust
fn _vectors_svecf32_aggregate_avg_sum_combinefunc(
    mut state1: Option<Internal>,
    mut state2: Option<Internal>,
) -> Option<Internal> {
    match (
        get_mut_internal::<SVecf32AggregateAvgSumStype>(&mut state1),
        get_mut_internal::<SVecf32AggregateAvgSumStype>(&mut state2),
    ) {
```

Reply:
> May I mark this function unsafe? Inline it would make this `match` statement to long.
> 
> ```rust
> fn _vectors_svecf32_aggregate_avg_sum_combinefunc(
>     mut state1: Option<Internal>,
>     mut state2: Option<Internal>,
> ) -> Option<Internal> {
>     match (
>         get_mut_internal::<SVecf32AggregateAvgSumStype>(&mut state1),
>         get_mut_internal::<SVecf32AggregateAvgSumStype>(&mut state2),
>     ) {
> ```

Reply:
> Inline it would make this match statement to long.

You can use `let ...` first.